
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../bio-Gene-expression/">
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.4.4">
    
    
      
        <title>Transcriptome Assembly - Roberts Lab Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.bd3936ea.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.356b1318.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#qc" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Roberts Lab Handbook" class="md-header__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Roberts Lab Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Transcriptome Assembly
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Roberts Lab Handbook" class="md-nav__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Roberts Lab Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Lab Culture
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Lab Culture
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Onboarding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Onboarding
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Code-of-Conduct/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code of Conduct
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Environment-and-Expectations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Environment and Expectations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Communication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../DEI-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DEI Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../External-Communication-and-Funding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Safety/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Safety
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Undergraduate-contract/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Undergraduate Contract
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing and Code
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Computing and Code
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Best-Practices/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Best Practices
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../code_Snippets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Hardware/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Computing Hardware
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
        
          
          <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Mox Guides
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_5">
            <span class="md-nav__icon md-icon"></span>
            Mox Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Adding-a-User/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    New User
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Logging-In/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Logging in
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Node-Types/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Node Types
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Data-Storage-and-System-Organization/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Storage and System Organization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Running-a-Job/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Running a job
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Viewing-and-Interacting-with-Jobs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Viewing and interacting with jobs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_File-Transfers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    File Transfers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Installing-Programs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installing software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Jupyter-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Jupyter Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_RStudio-Server/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RStudio Server
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Databases and Catalogs
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Databases and Catalogs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Inventory/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Inventory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Pubathon/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pubathon
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Digital-Media/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Digital Media
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Experiment-Database/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Experiment Database
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Genomic-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomic Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../music/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spotify Playlists
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Protocols
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Purchasing-and-Reimbursement/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Purchasing and Reimbursement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Data-Management/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data-Management
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Protocols
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Chemical-Standard-Operating-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Chemical SOPs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Alarm-Information/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Alarm Information
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Bioinformatic Workflows
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Bioinformatic Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Basics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_DNA-methylation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Gene-expression/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene Expression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Transcriptome Assembly
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Transcriptome Assembly
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#qc" class="md-nav__link">
    QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming" class="md-nav__link">
    Trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    Assembly
  </a>
  
    <nav class="md-nav" aria-label="Assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sample-list-file" class="md-nav__link">
    Sample list file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stranded-sequencing-reads" class="md-nav__link">
    Stranded sequencing reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#de-novo-assembly" class="md-nav__link">
    De novo assembly
  </a>
  
    <nav class="md-nav" aria-label="De novo assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-guided-assembly" class="md-nav__link">
    Genome-guided assembly
  </a>
  
    <nav class="md-nav" aria-label="Genome-guided assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_1" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    Output files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-expression" class="md-nav__link">
    Gene expression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation" class="md-nav__link">
    Annotation
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#qc" class="md-nav__link">
    QC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trimming" class="md-nav__link">
    Trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    Assembly
  </a>
  
    <nav class="md-nav" aria-label="Assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sample-list-file" class="md-nav__link">
    Sample list file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#stranded-sequencing-reads" class="md-nav__link">
    Stranded sequencing reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#de-novo-assembly" class="md-nav__link">
    De novo assembly
  </a>
  
    <nav class="md-nav" aria-label="De novo assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-guided-assembly" class="md-nav__link">
    Genome-guided assembly
  </a>
  
    <nav class="md-nav" aria-label="Genome-guided assembly">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_1" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    Output files
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-expression" class="md-nav__link">
    Gene expression
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation" class="md-nav__link">
    Annotation
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/RobertsLab/resources/blob/master/docs/bio_Transcriptome-assembly.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
    </a>
  
  


  <h1>Transcriptome Assembly</h1>

<p>Introductory guide to transcriptome assembly using <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> and short-read sequencing data.</p>
<h2 id="qc">QC</h2>
<p>A normal starting point would be having raw sequence data provided by a core facility. When downloading said data you need to make sure you check the integrity of the files after transfer by confirming checksum hashes (usually <code>MD5</code> checksums) match those provided by the sequencing facility. </p>
<p>You should run <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a> to assess sequencing quality. This might vary based on the details of your project but generally you can ID outliers and those samples with poor read quality. Presence of adapters can also be visualized.</p>
<h2 id="trimming">Trimming</h2>
<p>Quality and adapter trimming is required prior to assembly. <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> is recommended due to its speed and FastQC-like report(s). Additionally, the output from <a href="https://github.com/OpenGene/fastp"><code>fastp</code></a> can also be analyzed by <a href="https://multiqc.info/"><code>MultiQC</code></a> (requires a "plug-in"). Trimmed files should be passed through <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><code>FastQC</code></a> and assessed. Although rare, some projects may require a second round of trimming.</p>
<p>Alternatively, <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> has trimming capabilities built in, using <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>. Although convenient, it limits the ability to assess post-trimming sequencing data prior to assembly.</p>
<h2 id="assembly">Assembly</h2>
<p><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> is the <em>de facto</em> standard. It is well-documented, well-supported, and actively updated. Additionally, the developer is very responsive, considerate, and helpful to <a href="https://github.com/trinityrnaseq/trinityrnaseq/issues">all GitHub Issues</a>.</p>
<p><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> is powerful and has complex, but useful options availalbe. Take time to consider how you will use your assembly for later analysis. <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> has many options available for downstream analysis (e.g. <a href="#gene-expression">gene expression</a>) that can be simplified with careful planning prior to assembly.</p>
<p>Due to the intensive processing required for assembly (high CPU and RAM usage), it is highly recommended to run all assemblies on an <a href="https://robertslab.github.io/resources/mox_Node-Types/">execute node on Mox</a>. As such, all code examples are written with the assumption that the commands are being run on Mox.</p>
<h3 id="sample-list-file">Sample list file</h3>
<p>It is recommended to create a sample list file for <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> to use. One of the biggest benefits is that this sample file list can be used for other downstream operations in the <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> pipeline. Additionally, it's an easy way to document which sequencing files were used for assembly. Here's the example from <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a>. Sample file list is tab-delimited like this:</p>
<div class="highlight"><pre><span></span><code>cond_A    cond_A_rep1    A_rep1_left.fq    A_rep1_right.fq
cond_A    cond_A_rep2    A_rep2_left.fq    A_rep2_right.fq
cond_B    cond_B_rep1    B_rep1_left.fq    B_rep1_right.fq
cond_B    cond_B_rep2    B_rep2_left.fq    B_rep2_right.fq
</code></pre></div>
<h3 id="stranded-sequencing-reads">Stranded sequencing reads</h3>
<p><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> has the option (<code>--SS_lib_type</code>) to specify whether or not the sequences you're assembly are "stranded". This is dependent upon the library construction. With that said, most paired-end RNA-seq project libraries are constructed using Illumina's stranded kit. As such, the user should specify this in the following fashion as on option in the <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> command (example specifies typical stranded libraries): <code>--SS_lib_type RF</code></p>
<p>If you do not know whether your libraries are stranded or not (for example, if you downloaded RNA-seq data from NCBI and the metadata doesn't indicate library construction methodology), <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> has a built-in tool to help assess your sequencing reads, <em>after</em> assembly:</p>
<p><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki/Examine-Strand-Specificity">Examine-Strand-Specificity</a></p>
<h3 id="de-novo-assembly"><em>De novo</em> assembly</h3>
<p>A <em>de novo</em> assembly is an assembly that is done without the use of a reference genome. Here's an example command, using trimmed paired-end reads. This set of commands will assembly the reads into contigs, generate assembly statistics, a gene map file (maps isoforms to "gene" names), and a sequence length file (useful for downstream gene expression).</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Perform assembly</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/Trinity<span class="w"> </span><span class="se">\</span>
--seqType<span class="w"> </span>fq<span class="w"> </span><span class="se">\</span>
--SS_lib_type<span class="w"> </span>RF<span class="w"> </span><span class="se">\</span>
--max_memory<span class="w"> </span>100G<span class="w"> </span><span class="se">\</span>
--CPU<span class="w"> </span><span class="si">${</span><span class="nv">threads</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--samples_file<span class="w"> </span><span class="si">${</span><span class="nv">samples</span><span class="si">}</span>

<span class="c1"># Assembly stats</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/TrinityStats.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="si">${</span><span class="nv">assembly_stats</span><span class="si">}</span>

<span class="c1"># Create gene map files</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/support_scripts/get_Trinity_gene_to_trans_map.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span>.gene_trans_map

<span class="c1"># Create sequence lengths file (used for differential gene expression)</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/misc/fasta_seq_length.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span>.seq_lens
</code></pre></div>
<ul>
<li><code>--max_memory 100G</code> should <em>not</em> be changed, per communications with the developer.</li>
</ul>
<h4 id="use-cases-from-our-lab">Use cases from our lab</h4>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2020/03/30/Transcriptome-Assembly-C.bairdi-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox.html">Transcriptome-Assembly-C.bairdi-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox</a></li>
</ul>
<h3 id="genome-guided-assembly">Genome-guided assembly</h3>
<p>A genome-guided assembly is an assembly which utilizes a reference genome. This requires a sorted BAM as input, which means you have to have previously aligned your RNA-seq reads to a reference genome. See our Handbook entry on using <a href="https://robertslab.github.io/resources/bio-Gene-expression/#alignment-hisat2">Hisat2</a> for read alignment. Here's an example command, using trimmed paired-end reads. This set of commands will assembly the reads into contigs, generate assembly statistics, a gene map file (maps isoforms to "gene" names), and a sequence length file (useful for downstream gene expression).</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Perform assembly</span>
<span class="si">${</span><span class="nv">programs_array</span><span class="p">[trinity]</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--genome_guided_bam<span class="w"> </span><span class="si">${</span><span class="nv">sorted_bam</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--genome_guided_max_intron<span class="w"> </span><span class="si">${</span><span class="nv">max_intron</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--seqType<span class="w"> </span>fq<span class="w"> </span><span class="se">\</span>
--SS_lib_type<span class="w"> </span>RF<span class="w"> </span><span class="se">\</span>
--max_memory<span class="w"> </span>100GB<span class="w"> </span><span class="se">\</span>
--CPU<span class="w"> </span><span class="si">${</span><span class="nv">threads</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--samples_file<span class="w"> </span><span class="si">${</span><span class="nv">samples</span><span class="si">}</span>

<span class="c1"># Assembly stats</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/TrinityStats.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="si">${</span><span class="nv">assembly_stats</span><span class="si">}</span>

<span class="c1"># Create gene map files</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/support_scripts/get_Trinity_gene_to_trans_map.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span>.gene_trans_map

<span class="c1"># Create sequence lengths file (used for differential gene expression)</span>
<span class="si">${</span><span class="nv">trinity_dir</span><span class="si">}</span>/util/misc/fasta_seq_length.pl<span class="w"> </span><span class="se">\</span>
trinity_out_dir/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">fasta_name</span><span class="si">}</span><span class="s2">&quot;</span>.seq_lens
</code></pre></div>
<ul>
<li>
<p><code>--genome_guided_max_intron ${max_intron}</code>: The value used in the <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki"><code>Trinity</code></a> examples is 10000.</p>
</li>
<li>
<p><code>--max_memory 100G</code> should <em>not</em> be changed, per communications with the developer.</p>
</li>
</ul>
<h4 id="use-cases-from-our-lab_1">Use cases from our lab</h4>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/2022/02/07/Transcriptome-Assembly-Genome-guided-C.virginica-Adult-Gonad-OA-RNAseq-Using-Trinity-on-Mox.html">Transcriptome-Assembly-Genome-guided-C.virginica-Adult-Gonad-OA-RNAseq-Using-Trinity-on-Mox</a></li>
</ul>
<h2 id="output-files">Output files</h2>
<p>Both types of assemblies listed above will generate your assembly as a FastA file:</p>
<ul>
<li><code>Trinity.fasta</code>: This is the default name.</li>
</ul>
<p>If you ran the commands above you will also get the following files:</p>
<ul>
<li>
<p><code>assembly_stats.txt</code>: Statistics on your assembly. Will look something like this:</p>
<div class="highlight"><pre><span></span><code>################################
## Counts of transcripts, etc.
################################
Total trinity &#39;genes&#39;:  887315
Total trinity transcripts:  1849486
Percent GC: 36.26

########################################
Stats based on ALL transcript contigs:
########################################

Contig N10: 7967
Contig N20: 5284
Contig N30: 3814
Contig N40: 2801
Contig N50: 2062

Median contig length: 562
Average contig: 1117.46
Total assembled bases: 2066718534


#####################################################
## Stats based on ONLY LONGEST ISOFORM per &#39;GENE&#39;:
#####################################################

Contig N10: 6904
Contig N20: 4398
Contig N30: 3003
Contig N40: 2120
Contig N50: 1501

Median contig length: 434
Average contig: 860.79
Total assembled bases: 763788564
</code></pre></div>
</li>
<li>
<p><code>Trinity.fasta.gene_trans_map</code>:</p>
<div class="highlight"><pre><span></span><code>TRINITY_GG_1_c20044_g1  TRINITY_GG_1_c20044_g1_i2
TRINITY_GG_1_c20044_g1  TRINITY_GG_1_c20044_g1_i4
TRINITY_GG_1_c27757_g4  TRINITY_GG_1_c27757_g4_i1
TRINITY_GG_1_c4646_g1   TRINITY_GG_1_c4646_g1_i1
TRINITY_GG_1_c31636_g3  TRINITY_GG_1_c31636_g3_i1
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i2
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i7
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i5
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i6
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i4
TRINITY_GG_1_c5375_g1   TRINITY_GG_1_c5375_g1_i1
</code></pre></div>
</li>
<li>
<p><code>Trinity.fasta.seq_lens</code>:</p>
<div class="highlight"><pre><span></span><code>#fasta_entry    length
TRINITY_GG_1_c20044_g1_i2   1058
TRINITY_GG_1_c20044_g1_i4   1057
TRINITY_GG_1_c27757_g4_i1   265
TRINITY_GG_1_c4646_g1_i1    347
TRINITY_GG_1_c31636_g3_i1   215
TRINITY_GG_1_c5375_g1_i2    324
TRINITY_GG_1_c5375_g1_i7    511
TRINITY_GG_1_c5375_g1_i5    349
TRINITY_GG_1_c5375_g1_i6    340
</code></pre></div>
</li>
</ul>
<h2 id="gene-expression">Gene expression</h2>
<h2 id="annotation">Annotation</h2>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit"], "search": "../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.94c44541.min.js"></script>
      
    
  </body>
</html>