
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../bio_Basics/">
      
      
        <link rel="next" href="../bio_DNA-methylation/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.4.4">
    
    
      
        <title>Annotation - Roberts Lab Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.bd3936ea.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.356b1318.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#annotation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Roberts Lab Handbook" class="md-header__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Roberts Lab Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Annotation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Roberts Lab Handbook" class="md-nav__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Roberts Lab Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Lab Culture
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Lab Culture
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Onboarding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Onboarding
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Code-of-Conduct/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code of Conduct
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Environment-and-Expectations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Environment and Expectations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Communication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../DEI-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DEI Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../External-Communication-and-Funding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Safety/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Safety
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Undergraduate-contract/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Undergraduate Contract
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing and Code
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Computing and Code
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Best-Practices/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Best Practices
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../code_Snippets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Hardware/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Computing Hardware
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
        
          
          <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Mox Guides
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_5">
            <span class="md-nav__icon md-icon"></span>
            Mox Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Adding-a-User/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    New User
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Logging-In/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Logging in
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Node-Types/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Node Types
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Data-Storage-and-System-Organization/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Storage and System Organization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Running-a-Job/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Running a job
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Viewing-and-Interacting-with-Jobs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Viewing and interacting with jobs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_File-Transfers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    File Transfers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Installing-Programs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installing software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Jupyter-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Jupyter Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_RStudio-Server/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RStudio Server
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Databases and Catalogs
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Databases and Catalogs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Inventory/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Inventory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Pubathon/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pubathon
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Digital-Media/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Digital Media
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Experiment-Database/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Experiment Database
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Genomic-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomic Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../music/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spotify Playlists
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Protocols
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Purchasing-and-Reimbursement/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Purchasing and Reimbursement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Data-Management/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data-Management
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Protocols
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Chemical-Standard-Operating-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Chemical SOPs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Alarm-Information/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Alarm Information
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Bioinformatic Workflows
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Bioinformatic Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Basics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Annotation
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Annotation
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#intro-blast" class="md-nav__link">
    Intro (Blast)
  </a>
  
    <nav class="md-nav" aria-label="Intro (Blast)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-notebooks" class="md-nav__link">
    Blast Notebooks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-ontology-go" class="md-nav__link">
    Gene Ontology (GO)
  </a>
  
    <nav class="md-nav" aria-label="Gene Ontology (GO)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-go-terms-from-uniprot-using-swissprot-ids" class="md-nav__link">
    Retrieve GO terms from UniProt Using SwissProt IDs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genome-features" class="md-nav__link">
    Genome features
  </a>
  
    <nav class="md-nav" aria-label="Genome features">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bedtoolsintersectbed" class="md-nav__link">
    bedtools::intersectbed
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#transcriptome-trinity" class="md-nav__link">
    Transcriptome (Trinity)
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_DNA-methylation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Gene-expression/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene Expression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Transcriptome-assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Transcriptome Assembly
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#intro-blast" class="md-nav__link">
    Intro (Blast)
  </a>
  
    <nav class="md-nav" aria-label="Intro (Blast)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-notebooks" class="md-nav__link">
    Blast Notebooks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-ontology-go" class="md-nav__link">
    Gene Ontology (GO)
  </a>
  
    <nav class="md-nav" aria-label="Gene Ontology (GO)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-go-terms-from-uniprot-using-swissprot-ids" class="md-nav__link">
    Retrieve GO terms from UniProt Using SwissProt IDs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genome-features" class="md-nav__link">
    Genome features
  </a>
  
    <nav class="md-nav" aria-label="Genome features">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bedtoolsintersectbed" class="md-nav__link">
    bedtools::intersectbed
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#transcriptome-trinity" class="md-nav__link">
    Transcriptome (Trinity)
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/RobertsLab/resources/blob/master/docs/bio-Annotation.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
    </a>
  
  


<h1 id="annotation">Annotation</h1>
<h2 id="intro-blast">Intro (Blast)</h2>
<p>Blast is a key component of working with lesser studied taxa. Here are some resources to help with this.</p>
<p><strong>First off, you should be familar with the command line interface and bash</strong></p>
<ul>
<li><a href="http://swcarpentry.github.io/shell-novice/01-intro/">Introducing the Shell</a></li>
<li><a href="https://datacarpentry.org/shell-genomics/">Introduction to the Command Line for Genomics</a></li>
<li><a href="https://explainshell.com/">https://explainshell.com/</a></li>
</ul>
<hr />
<h3 id="blast-notebooks">Blast Notebooks</h3>
<ul>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/09-blast.ipynb">https://github.com/RobertsLab/code/blob/master/09-blast.ipynb</a> - An example of how one might take a multi sequence fasta file and using NCBI Blast, compare the sequences with the Swiss-Prot Database on their own computer.</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/10-blast-2-slim.ipynb">https://github.com/RobertsLab/code/blob/master/10-blast-2-slim.ipynb</a> - A notebook to seamlessly take blast output to GO Slim list</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/script-box/complete_go_annotation_notebook.Rmd">https://github.com/RobertsLab/code/blob/master/script-box/complete_go_annotation_notebook.Rmd</a> -</p>
</li>
<li>
<p>https://github.com/sr320/ceabigr/blob/main/code/17-Swiss-Prot-Annotation.Rmd -  Blasting C virginica to Swiss-Prot. Author: Steven Roberts <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/ceabigr" /></p>
</li>
</ul>
<hr />
<h2 id="gene-ontology-go">Gene Ontology (GO)</h2>
<h3 id="retrieve-go-terms-from-uniprot-using-swissprot-ids">Retrieve GO terms from UniProt Using SwissProt IDs</h3>
<p>The following steps will use the UniProt Python API to create a tab-delimited file of data retrieved from UniProt.</p>
<ol>
<li>
<p>Create newline-delimited file of SwissProt IDs. (e.g. <code>SPIDS.txt</code>)</p>
<div class="highlight"><pre><span></span><code>cat<span class="w"> </span>SPIDS.txt

Q86IC9
P04177
Q8L840
Q61043
A1E2V0
P34456
P34457
O00463
Q00945
Q5SWK7
</code></pre></div>
</li>
<li>
<p>Create Python file (e.g. <code>uniprot-retrieval.py</code>) with the following:</p>
<div class="highlight"><pre><span></span><code><span class="kn">import</span> <span class="nn">re</span>
<span class="kn">import</span> <span class="nn">zlib</span>
<span class="kn">import</span> <span class="nn">gzip</span>
<span class="kn">import</span> <span class="nn">requests</span>
<span class="kn">from</span> <span class="nn">requests.adapters</span> <span class="kn">import</span> <span class="n">HTTPAdapter</span><span class="p">,</span> <span class="n">Retry</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">shutil</span>

<span class="n">re_next_link</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="sa">r</span><span class="s1">&#39;&lt;(.+)&gt;; rel=&quot;next&quot;&#39;</span><span class="p">)</span>
<span class="n">retries</span> <span class="o">=</span> <span class="n">Retry</span><span class="p">(</span><span class="n">total</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">backoff_factor</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">status_forcelist</span><span class="o">=</span><span class="p">[</span><span class="mi">500</span><span class="p">,</span> <span class="mi">502</span><span class="p">,</span> <span class="mi">503</span><span class="p">,</span> <span class="mi">504</span><span class="p">])</span>
<span class="n">session</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">Session</span><span class="p">()</span>
<span class="n">session</span><span class="o">.</span><span class="n">mount</span><span class="p">(</span><span class="s2">&quot;https://&quot;</span><span class="p">,</span> <span class="n">HTTPAdapter</span><span class="p">(</span><span class="n">max_retries</span><span class="o">=</span><span class="n">retries</span><span class="p">))</span>


<span class="k">def</span> <span class="nf">get_next_link</span><span class="p">(</span><span class="n">headers</span><span class="p">):</span>
    <span class="k">if</span> <span class="s2">&quot;Link&quot;</span> <span class="ow">in</span> <span class="n">headers</span><span class="p">:</span>
        <span class="n">match</span> <span class="o">=</span> <span class="n">re_next_link</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">headers</span><span class="p">[</span><span class="s2">&quot;Link&quot;</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>


<span class="k">def</span> <span class="nf">get_batch</span><span class="p">(</span><span class="n">batch_url</span><span class="p">):</span>
    <span class="k">while</span> <span class="n">batch_url</span><span class="p">:</span>
        <span class="n">response</span> <span class="o">=</span> <span class="n">session</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">batch_url</span><span class="p">)</span>
        <span class="n">response</span><span class="o">.</span><span class="n">raise_for_status</span><span class="p">()</span>
        <span class="n">total</span> <span class="o">=</span> <span class="n">response</span><span class="o">.</span><span class="n">headers</span><span class="p">[</span><span class="s2">&quot;x-total-results&quot;</span><span class="p">]</span>
        <span class="k">yield</span> <span class="n">response</span><span class="p">,</span> <span class="n">total</span>
        <span class="n">batch_url</span> <span class="o">=</span> <span class="n">get_next_link</span><span class="p">(</span><span class="n">response</span><span class="o">.</span><span class="n">headers</span><span class="p">)</span>


<span class="k">def</span> <span class="nf">process_accessions</span><span class="p">(</span><span class="n">accessions</span><span class="p">):</span>
    <span class="n">accession_batches</span> <span class="o">=</span> <span class="p">[</span><span class="n">accessions</span><span class="p">[</span><span class="n">i</span><span class="p">:</span><span class="n">i</span><span class="o">+</span><span class="mi">500</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">accessions</span><span class="p">),</span> <span class="mi">500</span><span class="p">)]</span>
    <span class="n">all_lines</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">accession_batch</span> <span class="ow">in</span> <span class="n">accession_batches</span><span class="p">:</span>
        <span class="n">accession_query</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%29%</span><span class="s1">20OR</span><span class="si">%20%</span><span class="s1">28accession%3A&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">accession_batch</span><span class="p">)</span>
        <span class="n">url</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;https://rest.uniprot.org/uniprotkb/search?compressed=true&amp;fields=accession%2Creviewed%2Cid%2Cprotein_name%2Cgene_names%2Corganism_name%2Clength%2Cgo_p%2Cgo_c%2Cgo%2Cgo_f%2Cgo_id&amp;format=tsv&amp;query=%28%28accession%3A</span><span class="si">{</span><span class="n">accession_query</span><span class="si">}</span><span class="s2">%29%29&amp;size=500&quot;</span>

        <span class="n">progress</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">lines</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">batch</span><span class="p">,</span> <span class="n">total</span> <span class="ow">in</span> <span class="n">get_batch</span><span class="p">(</span><span class="n">url</span><span class="p">):</span>
            <span class="c1"># Decompress each batch as we want to extract the header</span>
            <span class="n">decompressed</span> <span class="o">=</span> <span class="n">zlib</span><span class="o">.</span><span class="n">decompress</span><span class="p">(</span><span class="n">batch</span><span class="o">.</span><span class="n">content</span><span class="p">,</span> <span class="mi">16</span> <span class="o">+</span> <span class="n">zlib</span><span class="o">.</span><span class="n">MAX_WBITS</span><span class="p">)</span>
            <span class="n">batch_lines</span> <span class="o">=</span> <span class="p">[</span><span class="n">line</span> <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">decompressed</span><span class="o">.</span><span class="n">decode</span><span class="p">(</span><span class="s2">&quot;utf-8&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span> <span class="k">if</span> <span class="n">line</span><span class="p">]</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="n">progress</span><span class="p">:</span>
                <span class="c1"># First line so print TSV header</span>
                <span class="n">lines</span> <span class="o">=</span> <span class="p">[</span><span class="n">batch_lines</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
            <span class="n">lines</span> <span class="o">+=</span> <span class="n">batch_lines</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span>
            <span class="n">progress</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">lines</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">progress</span><span class="si">}</span><span class="s2"> / </span><span class="si">{</span><span class="n">total</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="n">all_lines</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">lines</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">all_lines</span>


<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">2</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Usage: python uniprot-retrieval.py &lt;input_file&gt;&quot;</span><span class="p">)</span>
        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

    <span class="n">accession_file</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">accession_file</span><span class="p">,</span> <span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">accessions</span> <span class="o">=</span> <span class="n">f</span><span class="o">.</span><span class="n">read</span><span class="p">()</span><span class="o">.</span><span class="n">splitlines</span><span class="p">()</span>

    <span class="n">retrieved_data</span> <span class="o">=</span> <span class="n">process_accessions</span><span class="p">(</span><span class="n">accessions</span><span class="p">)</span>

    <span class="c1"># Write to a temporary gzip file</span>
    <span class="n">temp_filename</span> <span class="o">=</span> <span class="s2">&quot;uniprot-retrieval-temp.tsv.gz&quot;</span>
    <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="s2">&quot;wt&quot;</span><span class="p">,</span> <span class="n">encoding</span><span class="o">=</span><span class="s2">&quot;utf-8&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">retrieved_data</span><span class="p">))</span>

    <span class="c1"># Merge the temporary file with the existing output file (if it exists)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s2">&quot;uniprot-retrieval.tsv.gz&quot;</span><span class="p">,</span> <span class="s2">&quot;rb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_existing</span><span class="p">,</span> <span class="nb">open</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="s2">&quot;rb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_temp</span><span class="p">:</span>
            <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s2">&quot;uniprot-retrieval-merged.tsv.gz&quot;</span><span class="p">,</span> <span class="s2">&quot;wb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_merged</span><span class="p">:</span>
                <span class="n">shutil</span><span class="o">.</span><span class="n">copyfileobj</span><span class="p">(</span><span class="n">f_existing</span><span class="p">,</span> <span class="n">f_merged</span><span class="p">)</span>
                <span class="n">shutil</span><span class="o">.</span><span class="n">copyfileobj</span><span class="p">(</span><span class="n">f_temp</span><span class="p">,</span> <span class="n">f_merged</span><span class="p">)</span>

        <span class="c1"># Replace the original output file with the merged file</span>
        <span class="n">shutil</span><span class="o">.</span><span class="n">move</span><span class="p">(</span><span class="s2">&quot;uniprot-retrieval-merged.tsv.gz&quot;</span><span class="p">,</span> <span class="s2">&quot;uniprot-retrieval.tsv.gz&quot;</span><span class="p">)</span>
    <span class="k">except</span> <span class="ne">FileNotFoundError</span><span class="p">:</span>
        <span class="c1"># If the existing output file doesn&#39;t exist, rename the temporary file</span>
        <span class="n">shutil</span><span class="o">.</span><span class="n">move</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="s2">&quot;uniprot-retrieval.tsv.gz&quot;</span><span class="p">)</span>
</code></pre></div>
</li>
<li>
<p>Run the Python script:</p>
<p><div class="highlight"><pre><span></span><code>python3<span class="w"> </span>uniprot-retrieval.py<span class="w"> </span>SPIDS.txt
</code></pre></div>
- IMPORTANT: Requires Python &gt;= 3!</p>
<div class="highlight"><pre><span></span><code>- If using R Markdown, run the above in a `bash` chunk:
</code></pre></div>
</li>
</ol>
<p>The resulting output file (<code>uniprot-retrieval.tsv.gz</code>) will be in your working directory.</p>
<ol>
<li>
<p>Gunzip the output file:</p>
<div class="highlight"><pre><span></span><code>gunzip<span class="w"> </span>uniprot-retrieval.tsv.gz
</code></pre></div>
</li>
</ol>
<p>The resulting file (<code>uniprot-retrieval.tsv</code>) will be formatted with the following columns:</p>
<table>
<thead>
<tr>
<th>Entry</th>
<th>Reviewed</th>
<th>Entry Name</th>
<th>Protein names</th>
<th>Gene Names</th>
<th>Organism</th>
<th>Length</th>
<th>Gene Ontology (biological process)</th>
<th>Gene Ontology (cellular component)</th>
<th>Gene Ontology (molecular function)</th>
<th>Gene Ontology (GO)</th>
<th>Gene Ontology IDs</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>NOTES:</p>
<ul>
<li>
<p>This requires Python &gt;= 3 to run. Simplest way to access Python 3 is via a conda environment.</p>
</li>
<li>
<p>On the first attempt, you'll likely need to install the packages that are being imported at the very beginning of the script.</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/resources/issues">Create an issue</a> if you need help with any of the above.</p>
</li>
</ul>
<hr />
<h2 id="genome-features">Genome features</h2>
<p>In addition to sequence database alignment, finding spatial relationship within a genome is also an import approach for annotation. Often this is done using software tools such as <code>bedtools</code>.</p>
<h3 id="bedtoolsintersectbed"><code>bedtools::intersectbed</code></h3>
<p>see also https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html</p>
<hr />
<h2 id="transcriptome-trinity">Transcriptome (Trinity)</h2>
<p>After <a href="https://robertslab.github.io/resources/bio_Transcriptome-assembly/">transcriptome assembly using Trinity</a>, run the numbered steps below, in order.</p>
<p>NOTE: The following info is long and requires the use of many programs. All of the code listed below are solely examples. Making the commands functional requires a fair amount of organization (i.e. listing paths to programs and input/output files, creating subdirectories for organizing outputs, etc.). See the Use Cases at the end of this section for a more complete picture of how to organize/run this pipeline.</p>
<ol>
<li>
<p>Transdecoder</p>
<ul>
<li>
<p>Identify longest open reading frames (ORFs).</p>
</li>
<li>
<p>Use transcriptome assembly and gene-trans map from <a href="https://robertslab.github.io/resources/bio_Transcriptome-assembly/">Trinity assembly</a>.</p>
<div class="highlight"><pre><span></span><code>TransDecoder.LongOrfs \
--gene_trans_map Trinity.fasta.gene_trans_map \
-t Trinity.fasta&quot;
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>BLASTp</p>
<ul>
<li>
<p>Run blastp on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Output format 6 produces a standard BLAST tab-delimited file.</p>
</li>
<li>
<p>Settings are recommended in TransDecoder documentation.</p>
</li>
<li>
<p>Peptide database (<code>-db uniprot_sprot.pep</code>) is supplied with Trinotate (e.g. <code>Trinotate-v3.1.1/admin/uniprot_sprot.pep</code>), but can be changed to use your own version.</p>
<div class="highlight"><pre><span></span><code>blastp \
-query longest_orfs.pep \
-db uniprot_sprot.pep \
-max_target_seqs 1 \
-outfmt 6 \
-evalue 1e-5 \
-num_threads ${threads} \
&gt; Trinity.fasta.blastp.outfmt6
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>BLASTx (<a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a>)</p>
<ul>
<li>
<p>Run <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a> blastx on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Output format 6 produces a standard BLAST tab-delimited file.</p>
</li>
<li>
<p>Settings (<code>--evalue</code> and <code>--max-target-seqs</code>) are recommended in TransDecoder documentation.</p>
</li>
<li>
<p><code>--block-size</code> and <code>--index-chunks</code> are specific to running <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a> BLASTx.</p>
</li>
<li>
<p><code>--db uniprot_sprot.dmnd</code> is a <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a>-formatted BLAST database. User can generate their own.</p>
<div class="highlight"><pre><span></span><code>```
diamond blastx \
--db uniprot_sprot.dmnd \
--query Trinity.fasta \
--out Trinity.fasta.blastx.outfmt6 \
--outfmt 6 \
--evalue 1e-4 \
--max-target-seqs 1 \
--block-size 15.0 \
--index-chunks 4
```
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>pFam</p>
<ul>
<li>
<p>Run pfam search on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Protein hidden Markov model database (<code>Pfam-A.hmm</code>) is supplied with Trinotate (e.g. <code>Trinotate-v3.1.1/admin/Pfam-A.hmm</code>), but can be changed to use your own version.</p>
<div class="highlight"><pre><span></span><code>hmmscan \
--cpu ${threads} \
--domtblout Trinity.fasta.pfam.domtblout \
Pfam-A.hmm \
longest_orfs.pep
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>Transdecoder</p>
<ul>
<li>Run Transdecoder using transcriptome assembly FastA, <code>blastp</code> and <code>Pfam</code> results.</li>
</ul>
<div class="highlight"><pre><span></span><code>TransDecoder.Predict \
    -t Trinity.fasta \
    --retain_pfam_hits Trinity.fasta.pfam.domtblout \
    --retain_blastp_hits Trinity.fasta.blastp.outfmt6
</code></pre></div>
</li>
<li>
<p>Trinotate</p>
<ul>
<li>
<p>Trinotate requires a large number of steps and programs!</p>
<ul>
<li>
<p>Run <code>signalp</code></p>
<div class="highlight"><pre><span></span><code>signalp \
-f short \
-n Trinity.fasta.trinotate.signalp.out \
longest_orfs.pep
</code></pre></div>
</li>
<li>
<p>Run <code>tmHMM</code></p>
<div class="highlight"><pre><span></span><code>tmhmm \
--short \
&lt; longest_orfs.pep \
&gt; Trinity.fasta.trinotate.tmhmm.out
</code></pre></div>
</li>
<li>
<p>Run <code>RNAmmer</code></p>
<ul>
<li>Uses a special Trinotate implementation of <code>rnammer</code> (e.g. <code>Trinotate/util/rnammer_support/RnammerTranscriptome.pl</code>)</li>
</ul>
<div class="highlight"><pre><span></span><code>RnammerTranscriptome.pl \
--transcriptome Trinity.fasta \
--path_to_rnammer rnammer
</code></pre></div>
</li>
<li>
<p>Load transcripts and coding regions into Trinotate SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
init \
--gene_trans_map Trinity.fasta.gene_trans_map \
--transcript_fasta Tinity.fasta \
--transdecoder_pep longest_orfs.pep
</code></pre></div>
</li>
<li>
<p>Load BLASTp/x homologies into SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_swissprot_blastp \
Trinity.fasta.blastp.outfmt6
</code></pre></div>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_swissprot_blastx \
Trinity.fasta.blastx.outfmt6
</code></pre></div>
</li>
<li>
<p>Load Pfam into SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_pfam \
Trinity.fasta.pfam.domtblout
</code></pre></div>
</li>
<li>
<p>Load transmembrane domains</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_tmhmm \
Trinity.fasta.trinotate.tmhmm.out
</code></pre></div>
</li>
<li>
<p>Load signal peptides</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_signalp \
Trinity.fasta.trinotate.signalp.out
</code></pre></div>
</li>
<li>
<p>Load RNAmmer</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_rnammer \
Trinity.fasta.rnammer.gff
</code></pre></div>
</li>
<li>
<p>Create annotation report</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
report \
&gt; Trinity.fasta.annotation_report.txt
</code></pre></div>
</li>
<li>
<p>Extract gene ontology (GO) terms from annotation report</p>
<div class="highlight"><pre><span></span><code>extract_GO_assignments_from_Trinotate_xls.pl \
--Trinotate_xls Trinity.fasta.annotation_report.txt \
-G \
--include_ancestral_terms \
&gt; Trinity.fasta.go_annotations.txt
</code></pre></div>
<ul>
<li>The output file is formatted like this (<code>&lt;trinity-id&gt;``&lt;tab&gt;``&lt;GO:NNNNNN,GO:NNNNN,...&gt;</code>):</li>
</ul>
<div class="highlight"><pre><span></span><code>TRINITY_DN0_c0_g1   GO:0003674,GO:0003824,GO:0003964,GO:0006139,GO:0006259,GO:0006310,GO:0006313,GO:0006725,GO:0006807,GO:0008150,GO:0008152,GO:0009987,GO:0016740,GO:0016772,GO:0016779,GO:0032196,GO:0034061,GO:0034641,GO:0043170,GO:0044237,GO:0044238,GO:0044260,GO:0044699,GO:0044710,GO:0044763,GO:0046483,GO:0071704,GO:0090304,GO:1901360
TRINITY_DN0_c10_g1  GO:0003674,GO:0003824,GO:0004659,GO:0004660,GO:0005488,GO:0005575,GO:0005829,GO:0005875,GO:0005965,GO:0006464,GO:0006807,GO:0008150,GO:0008152,GO:0008270,GO:0008318,GO:0009987,GO:0016740,GO:0016765,GO:0018342,GO:0018343,GO:0019538,GO:0032991,GO:0036211,GO:0043167,GO:0043169,GO:0043170,GO:0043234,GO:0043412,GO:0044237,GO:0044238,GO:0044260,GO:0044267,GO:0044422,GO:0044424,GO:0044430,GO:0044444,GO:0044446,GO:0044464,GO:0046872,GO:0046914,GO:0071704,GO:0097354,GO:1901564,GO:1902494,GO:1990234
TRINITY_DN0_c2_g4   GO:0000166,GO:0003674,GO:0005488,GO:0005524,GO:0005575,GO:0005737,GO:0005856,GO:0017076,GO:0030554,GO:0032553,GO:0032555,GO:0032559,GO:0035639,GO:0036094,GO:0043167,GO:0043168,GO:0043226,GO:0043228,GO:0043229,GO:0043232,GO:0044424,GO:0044464,GO:0097159,GO:0097367,GO:1901265,GO:1901363
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>Make transcript features annotation map</p>
<div class="highlight"><pre><span></span><code>```
Trinotate_get_feature_name_encoding_attributes.pl \
Trinity.fasta.annotation_report.txt \
&gt; Trinity.fasta.annotation_feature_map.txt
```
</code></pre></div>
</li>
</ul>
</li>
</ol>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit"], "search": "../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.94c44541.min.js"></script>
      
    
  </body>
</html>