
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../bio-Annotation/">
      
      
        <link rel="next" href="../bio-Gene-expression/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.4.4">
    
    
      
        <title>DNA Methylation Analysis - Roberts Lab Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.bd3936ea.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.356b1318.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bismark" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Roberts Lab Handbook" class="md-header__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Roberts Lab Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              DNA Methylation Analysis
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Roberts Lab Handbook" class="md-nav__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Roberts Lab Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Lab Culture
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Lab Culture
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Onboarding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Onboarding
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Code-of-Conduct/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code of Conduct
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Environment-and-Expectations/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Environment and Expectations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Communication/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../DEI-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DEI Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../External-Communication-and-Funding/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Communication
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Safety/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Safety
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Undergraduate-contract/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Undergraduate Contract
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing and Code
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Computing and Code
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Best-Practices/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Best Practices
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../code_Snippets/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Hardware/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Computing Hardware
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
        
          
          <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Mox Guides
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_5">
            <span class="md-nav__icon md-icon"></span>
            Mox Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Adding-a-User/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    New User
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Logging-In/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Logging in
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Node-Types/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Node Types
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Data-Storage-and-System-Organization/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Storage and System Organization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Running-a-Job/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Running a job
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Viewing-and-Interacting-with-Jobs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Viewing and interacting with jobs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_File-Transfers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    File Transfers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Software/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Installing-Programs/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installing software
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_Jupyter-Notebooks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Jupyter Notebooks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../mox_RStudio-Server/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    RStudio Server
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Databases and Catalogs
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Databases and Catalogs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Inventory/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Inventory
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Pubathon/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pubathon
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Digital-Media/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Digital Media
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Experiment-Database/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Experiment Database
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Genomic-Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Genomic Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../music/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spotify Playlists
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Protocols
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Purchasing-and-Reimbursement/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Purchasing and Reimbursement
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Data-Management/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data-Management
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lab Protocols
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Chemical-Standard-Operating-Protocols/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Chemical SOPs
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../Alarm-Information/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Alarm Information
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Bioinformatic Workflows
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Bioinformatic Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Basics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Annotation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Annotation
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#bismark" class="md-nav__link">
    Bismark
  </a>
  
    <nav class="md-nav" aria-label="Bismark">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diagram" class="md-nav__link">
    Diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#code-output-expectations" class="md-nav__link">
    Code Output Expectations
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bismark_1" class="md-nav__link">
    Bismark
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bs-snpr" class="md-nav__link">
    BS-Snpr
  </a>
  
    <nav class="md-nav" aria-label="BS-Snpr">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_1" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#epidiversesnp-nextflow-pipeline" class="md-nav__link">
    EpiDiverse/snp (Nextflow pipeline)
  </a>
  
    <nav class="md-nav" aria-label="EpiDiverse/snp (Nextflow pipeline)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-mox" class="md-nav__link">
    Instructions for running on Mox
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#epidiversewgbs-nextflow-pipeline" class="md-nav__link">
    EpiDiverse/wgbs (Nextflow pipeline)
  </a>
  
    <nav class="md-nav" aria-label="EpiDiverse/wgbs (Nextflow pipeline)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-raven" class="md-nav__link">
    Instructions for running on Raven
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-mox_1" class="md-nav__link">
    Instructions for running on Mox
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methylkit" class="md-nav__link">
    Methylkit
  </a>
  
    <nav class="md-nav" aria-label="Methylkit">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_2" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diagram_1" class="md-nav__link">
    Diagram
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#characterizing-gene-level-methylation" class="md-nav__link">
    Characterizing Gene Level Methylation
  </a>
  
    <nav class="md-nav" aria-label="Characterizing Gene Level Methylation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_3" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Gene-expression/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene Expression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Transcriptome-assembly/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Transcriptome Assembly
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#bismark" class="md-nav__link">
    Bismark
  </a>
  
    <nav class="md-nav" aria-label="Bismark">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diagram" class="md-nav__link">
    Diagram
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#code-output-expectations" class="md-nav__link">
    Code Output Expectations
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bismark_1" class="md-nav__link">
    Bismark
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bs-snpr" class="md-nav__link">
    BS-Snpr
  </a>
  
    <nav class="md-nav" aria-label="BS-Snpr">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_1" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#epidiversesnp-nextflow-pipeline" class="md-nav__link">
    EpiDiverse/snp (Nextflow pipeline)
  </a>
  
    <nav class="md-nav" aria-label="EpiDiverse/snp (Nextflow pipeline)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-mox" class="md-nav__link">
    Instructions for running on Mox
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#epidiversewgbs-nextflow-pipeline" class="md-nav__link">
    EpiDiverse/wgbs (Nextflow pipeline)
  </a>
  
    <nav class="md-nav" aria-label="EpiDiverse/wgbs (Nextflow pipeline)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-raven" class="md-nav__link">
    Instructions for running on Raven
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#instructions-for-running-on-mox_1" class="md-nav__link">
    Instructions for running on Mox
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methylkit" class="md-nav__link">
    Methylkit
  </a>
  
    <nav class="md-nav" aria-label="Methylkit">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_2" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diagram_1" class="md-nav__link">
    Diagram
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#characterizing-gene-level-methylation" class="md-nav__link">
    Characterizing Gene Level Methylation
  </a>
  
    <nav class="md-nav" aria-label="Characterizing Gene Level Methylation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab_3" class="md-nav__link">
    Use cases from our lab
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/RobertsLab/resources/blob/master/docs/bio_DNA-methylation.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
    </a>
  
  


  <h1>DNA Methylation Analysis</h1>

<h2 id="bismark">Bismark</h2>
<p>See also the official documentation.</p>
<p>User Guides</p>
<p>1)  <a href="http://felixkrueger.github.io/Bismark/Docs/">http://felixkrueger.github.io/Bismark/Docs/</a>\
2)  <a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">https://www.bioinformatics.babraham.ac.uk/projects/bismark/</a></p>
<h3 id="use-cases-from-our-lab">Use cases from our lab</h3>
<p><a href="https://github.com/RobertsLab/code/blob/master/20-bismark.sh">https://github.com/RobertsLab/code/blob/master/20-bismark.sh</a></p>
<ul>
<li>
<p><a href="https://github.com/sr320/paper-oly-mbdbs-gen/blob/master/code/00-Bismark.sh">https://github.com/sr320/paper-oly-mbdbs-gen/blob/master/code/00-Bismark.sh</a> - used to processes BS-MBDSeq Data from Olympia oysters, run on Mox. Author: Steven Roberts <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/paper-oly-mbdbs-gen" /></p>
</li>
<li>
<p><a href="https://raw.githubusercontent.com/laurahspencer/C.magister_methyl-oa/master/scripts/20201214_Cmag_bismark-align.sh">https://raw.githubusercontent.com/laurahspencer/C.magister_methyl-oa/master/scripts/20201214_Cmag_bismark-align.sh</a> - slurm script used to process MiSeq data from Dungeness crab, run on Mox. <a href="https://github.com/laurahspencer/C.magister_methyl-oa/blob/master/notebooks/MBD-01%20Processing%20QC%20MiSeq%20data.ipynb">Here is a Jupyter Notebook</a> with more details/narrative. Author: Laura Spencer, but derived from the <a href="https://github.com/hputnam/Meth_Compare">MethCompare workflow</a>. <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/laurahspencer/C.magister_methyl-oa" /></p>
</li>
<li>
<p><a href="https://github.com/sr320/paper-oly-wgbs/blob/master/submission/Narrative.Rmd">https://github.com/sr320/paper-oly-wgbs/blob/master/submission/Narrative.Rmd</a> part of Rmd narrative, used for WGBS Olympia oyster data. Author: Steven Roberts <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/paper-oly-wgbs" /></p>
</li>
<li>
<p><a href="https://github.com/hputnam/Geoduck_Meth/blob/master/code/03-bismark.sh">https://github.com/hputnam/Geoduck_Meth/blob/master/code/03-bismark.sh</a> geoduck environmental memory project. Run on Mox. Author: Steven Roberts <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/hputnam/Geoduck_Meth" /></p>
</li>
<li>
<p><a href="https://raw.githubusercontent.com/epigeneticstoocean/paper-gonad-meth/master/code/02-bismark.sh">https://raw.githubusercontent.com/epigeneticstoocean/paper-gonad-meth/master/code/02-bismark.sh</a> eastern oyster data, run on Mox <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/epigeneticstoocean/paper-gonad-meth" /></p>
</li>
<li>
<p><a href="https://github.com/hputnam/Meth_Compare/blob/master/code/00.01-DNA-sequence-processing.md">https://github.com/hputnam/Meth_Compare/blob/master/code/00.01-DNA-sequence-processing.md</a> Complete DNA processing protocol from comparison of BS methods on corals. <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/hputnam/Meth_Compare" /></p>
</li>
</ul>
<h3 id="diagram">Diagram</h3>
<h3 id="code-output-expectations">Code Output Expectations</h3>
<p>(<strong>Always default to the Manual/User Guide!</strong> - this is merely an attempt at explaining our workflow)</p>
<h2 id="bismark_1">Bismark</h2>
<p><a href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.htm">Bismark User Guide</a></p>
<p><strong>(I) Running bismark_genome_preparation</strong></p>
<p>USAGE: bismark_genome_preparation [options] <path_to_genome_folder></p>
<div class="highlight"><pre><span></span><code>${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
${genome_folder}
</code></pre></div>
<p>You should expect to prepared genome with directory structure similar to</p>
<div class="highlight"><pre><span></span><code>./roslin_M/Bisulfite_Genome
./roslin_M/Bisulfite_Genome/GA_conversion
./roslin_M/Bisulfite_Genome/CT_conversion
</code></pre></div>
<p><strong>(II) Running bismark</strong></p>
<p>USAGE: bismark [options] --genome <genome_folder> {-1 <mates1> -2 <mates2> | <singles>}</p>
<div class="highlight"><pre><span></span><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg
</code></pre></div>
<p>This will create bam files (sequence alignment files)</p>
<p><strong>(III) Running deduplicate_bismark</strong></p>
<p>deduplicate_bismark --bam [options] <filenames></p>
<p>This command will deduplicate the Bismark alignment BAM file and remove all reads but one which align to the the very same position and in the same orientation. This step is recommended for whole-genome bisulfite samples, but should not be used for reduced representation libraries such as RRBS, amplicon or target enrichment libraries.</p>
<div class="highlight"><pre><span></span><code>find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam
</code></pre></div>
<p>This will create a deduplicated bam file.</p>
<p><strong>(IV) Running bismark_methylation_extractor</strong></p>
<p>USAGE: bismark_methylation_extractor [options] <filenames></p>
<div class="highlight"><pre><span></span><code>${bismark_dir}/bismark_methylation_extractor \
--bedGraph --counts --scaffolds \
--multicore 14 \
--buffer_size 75% \
*deduplicated.bam
</code></pre></div>
<div class="highlight"><pre><span></span><code># ${bismark_dir}/bismark_methylation_extractor \
# --bedGraph \
# --counts \
# --comprehensive \
# --merge_non_CpG \
# --multicore 28 \
# --buffer_size 75% \
# *deduplicated.bam
</code></pre></div>
<p>This will create deduplicated.bismark.cov.gz, uncompressed is in this format. Not we are using <code>--bedGraph</code> output, this is not default.</p>
<blockquote>
<p>Alternatively, the output of the methylation extractor can be transformed into a bedGraph and coverage file using the option --bedGraph (see also --counts)... Optionally, the bedGraph counts output can be used to generate a genome-wide cytosine report which reports the number on every single CpG (optionally every single cytosine) in the genome, irrespective of whether it was covered by any reads or not. As this type of report is informative for cytosines on both strands the output may be fairly large.</p>
</blockquote>
<div class="highlight"><pre><span></span><code>NC_035784.1 141 141 37.5    3   5
NC_035784.1 142 142 100 2   0
NC_035784.1 155 155 70  7   3
NC_035784.1 156 156 100 2   0
NC_035784.1 291 291 0   0   2
NC_035784.1 292 292 0   0   3
NC_035784.1 313 313 0   0   1
NC_035784.1 314 314 66.6666666666667    2   1
NC_035784.1 470 470 66.6666666666667    4   2
NC_035784.1 611 611 0   0   4
</code></pre></div>
<p><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></p>
<p><strong>genome-wide cytosine report output</strong></p>
<blockquote>
<p>Starting from the coverage output, the Bismark methylation extractor can optionally also output a genome-wide cytosine methylation report. The module coverage2cytosine (part of the Bismark package) may also be run individually. It is also sorted by chromosomal coordinates but also contains the sequence context and is in the following format: <chromosome> <position> <strand> <count methylated> <count unmethylated> <C-context> <trinucleotide context></p>
<p>The main difference to the bedGraph or coverage output is that every cytosine on both the top and bottom strands will be considered irrespective of whether they were actually covered by any reads in the experiment or not. For this to work one has to also specify the genome that was used for the Bismark alignments using the option --genome_folder <path>. As for the bedGraph mode, this will only consider cytosines in CpG context by default but can be extended to cytosines in any sequence context by using the option --CX (cf. Appendix (III)). Be aware though that this might mean an output with individual lines for more than 1.1 billion cytosines for any large mammalian genome...</p>
</blockquote>
<div class="highlight"><pre><span></span><code>find *deduplicated.bismark.cov.gz \
| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz
</code></pre></div>
<p>generates a file <code>.CpG_report.merged_CpG_evidence.cov</code></p>
<div class="highlight"><pre><span></span><code>NC_035785.1 217 219 100.000000  17  0
NC_035785.1 523 525 87.500000   7   1
NC_035785.1 556 558 50.000000   5   5
NC_035785.1 727 729 100.000000  16  0
NC_035785.1 1330    1332    0.000000    0   2
NC_035785.1 1403    1405    0.000000    0   2
NC_035785.1 1494    1496    66.666667   2   1
NC_035785.1 1747    1749    100.000000  8   0
NC_035785.1 2024    2026    100.000000  24  0
NC_035785.1 2054    2056    93.333333   14  1
</code></pre></div>
<p><strong>(V) Running bismark2report</strong></p>
<p>USAGE: bismark2report [options]</p>
<div class="highlight"><pre><span></span><code>${bismark_dir}/bismark2report
</code></pre></div>
<p><strong>(VI) Running bismark2summary</strong></p>
<p>USAGE: bismark2summary [options]</p>
<p>Produces report like <a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/PE_report.html">this</a></p>
<div class="highlight"><pre><span></span><code>${bismark_dir}/bismark2summary
</code></pre></div>
<p>Produces report like <a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/bismark_summary_report.html">this</a></p>
<hr />
<h2 id="bs-snpr">BS-Snpr</h2>
<p>See <a href="https://github.com/hellbelly/BS-Snper">https://github.com/hellbelly/BS-Snper</a></p>
<h3 id="use-cases-from-our-lab_1">Use cases from our lab</h3>
<ul>
<li><a href="https://nbviewer.org/github/RobertsLab/project-gigas-oa-meth/blob/master/code/07-BS-SNPer.ipynb">https://nbviewer.org/github/RobertsLab/project-gigas-oa-meth/blob/master/code/07-BS-SNPer.ipynb</a> - Pacific oyster exposed to OA. Author: Yaamini Venkataraman <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/RobertsLab/project-gigas-oa-meth" /></li>
</ul>
<hr />
<h2 id="epidiversesnp-nextflow-pipeline"><code>EpiDiverse/snp</code> (Nextflow pipeline)</h2>
<p>See <a href="https://github.com/EpiDiverse/snp">https://github.com/EpiDiverse/snp</a></p>
<h3 id="instructions-for-running-on-mox">Instructions for running on Mox</h3>
<p>Add the following below your SBATCH script header. Replace <code>bams_dir</code> and <code>genome_fasta</code> locations with your own.</p>
<p>NOTE: A FastA index file needs to be present in the same directory as your genome FastA file.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># These variables need to be set by user</span>

<span class="c1">## Directory with BAM(s)</span>
<span class="nv">bams_dir</span><span class="o">=</span><span class="s2">&quot;/gscratch/scrubbed/samwhite/data/C_virginica/BSseq/120321-cvBS&quot;</span>

<span class="c1">## Location of EpiDiverse/snp pipeline directory</span>
<span class="nv">epi_snp</span><span class="o">=</span><span class="s2">&quot;/gscratch/srlab/programs/epidiverse-pipelines/snp&quot;</span>

<span class="c1">## FastA file is required to end with .fa</span>
<span class="c1">## Requires FastA index file to be present in same directory as FastA</span>
<span class="nv">genome_fasta</span><span class="o">=</span><span class="s2">&quot;/gscratch/srlab/sam/data/C_virginica/genomes/GCF_002022765.2_C_virginica-3.0_genomic.fa&quot;</span>

<span class="c1">## Location of Nextflow</span>
<span class="nv">nextflow</span><span class="o">=</span><span class="s2">&quot;/gscratch/srlab/programs/nextflow-21.10.6-all&quot;</span>

<span class="c1">## Specify desired/needed version of Nextflow</span>
<span class="nv">nextflow_version</span><span class="o">=</span><span class="s2">&quot;20.07.1&quot;</span>


<span class="c1">###################################################################################</span>


<span class="c1"># Exit script if a command fails</span>
<span class="nb">set</span><span class="w"> </span>-e

<span class="c1"># Load Anaconda</span>
<span class="c1"># Uknown why this is needed, but Anaconda will not run if this line is not included.</span>
.<span class="w"> </span><span class="s2">&quot;/gscratch/srlab/programs/anaconda3/etc/profile.d/conda.sh&quot;</span>

<span class="c1"># Activate NF-core conda environment</span>
conda<span class="w"> </span>activate<span class="w"> </span>epidiverse-snp_env

<span class="c1"># Count BAMs</span>
<span class="c1"># Needed to pass info to Epidiverse/spn</span>
<span class="c1"># to avoid artificial file count limitation.</span>
<span class="nv">bam_count</span><span class="o">=</span><span class="m">0</span>

<span class="k">for</span><span class="w"> </span>bam<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="si">${</span><span class="nv">bams_dir</span><span class="si">}</span>*.bam
<span class="k">do</span>
<span class="w">  </span><span class="c1"># Increments counter by 1 for each BAM</span>
<span class="w">  </span><span class="o">((</span>bam_count++<span class="o">))</span>
<span class="k">done</span>

<span class="c1">## Run EpiDiverse/snp</span>
<span class="nv">NXF_VER</span><span class="o">=</span><span class="si">${</span><span class="nv">nextflow_version</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
<span class="si">${</span><span class="nv">nextflow</span><span class="si">}</span><span class="w"> </span>run<span class="w"> </span><span class="se">\</span>
<span class="si">${</span><span class="nv">epi_snp</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--input<span class="w"> </span><span class="si">${</span><span class="nv">bams_dir</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--reference<span class="w"> </span><span class="si">${</span><span class="nv">genome_fasta</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
--variants<span class="w"> </span><span class="se">\</span>
--clusters<span class="w"> </span><span class="se">\</span>
--take<span class="w"> </span><span class="si">${</span><span class="nv">bam_count</span><span class="si">}</span>
</code></pre></div>
<hr />
<h2 id="epidiversewgbs-nextflow-pipeline"><code>EpiDiverse/wgbs</code> (Nextflow pipeline)</h2>
<p>See <a href="https://github.com/EpiDiverse/wgbs">https://github.com/EpiDiverse/wgbs</a>.</p>
<h3 id="instructions-for-running-on-raven">Instructions for running on Raven</h3>
<h3 id="instructions-for-running-on-mox_1">Instructions for running on Mox</h3>
<p>NOTE: All code below should be added to your SLURM script.</p>
<ol>
<li>Add the following lines to the beginning (below the header) of your SLURM script:</li>
</ol>
<div class="highlight"><pre><span></span><code><span class="c1"># Load Anaconda</span>
<span class="c1"># Unknown why this is needed, but Anaconda will not run if this line is not included.</span>
.<span class="w"> </span><span class="s2">&quot;/gscratch/srlab/programs/anaconda3/etc/profile.d/conda.sh&quot;</span>


<span class="c1"># Activate the EpiDiverse/wbgs Anaconda environment</span>
conda<span class="w"> </span>activate<span class="w"> </span>epidiverse-wgbs_env
</code></pre></div>
<ol>
<li>
<p>Run the Nextflow pipeline. Read the comments in code below for important usage notes.</p>
<p>NOTE: Replace items enclosed in <code>&lt;&gt;</code> (including the <code>&lt;&gt;</code> with your own path(s))</p>
</li>
</ol>
<div class="highlight"><pre><span></span><code><span class="c1"># Run Nextflow EpiDiverse/wgbs pipeline</span>
<span class="c1"># Expects paired end, gzipped FastQ files named *.fastq.gz. Add --SE parameter to use single end instead.</span>
<span class="c1"># Genome FastA must have a corresponding FastA index file.</span>
<span class="c1"># Can perform trimming if desired. Add --trim parameter.</span>
<span class="c1"># Can run FastQC after trimming. Add --fastqc parameter.</span>
<span class="nv">NXF_VER</span><span class="o">=</span><span class="m">20</span>.07.1<span class="w"> </span><span class="se">\</span>
/gscratch/srlab/programs/nextflow<span class="w"> </span><span class="se">\</span>
/gscratch/srlab/programs/epidiverse-pipelines/wgbs<span class="w"> </span><span class="se">\</span>
--input<span class="w"> </span>&lt;path<span class="w"> </span>to<span class="w"> </span>directory<span class="w"> </span>with<span class="w"> </span>*.fastq.gz<span class="w"> </span>files&gt;<span class="w"> </span><span class="se">\</span>
--reference<span class="w"> </span>&lt;path<span class="w"> </span>to<span class="w"> </span>genome<span class="w"> </span>FastA&gt;<span class="w"> </span><span class="se">\</span>
--INDEX
</code></pre></div>
<h2 id="methylkit">Methylkit</h2>
<p>See also the official documentation - <a href="https://bioconductor.org/packages/release/bioc/vignettes/methylKit/inst/doc/methylKit.html">MethylKit Vignette</a></p>
<h3 id="use-cases-from-our-lab_2">Use cases from our lab</h3>
<ul>
<li>
<p><a href="https://github.com/sr320/paper-oly-mbdbs-gen/blob/master/code/01-methylkit.Rmd">https://github.com/sr320/paper-oly-mbdbs-gen/blob/master/code/01-methylkit.Rmd</a> - used to processes BS-MBDSeq Data from Olympia oysters, run on personal computer (not Mox). Author: Laura Spencer <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/paper-oly-mbdbs-gen" /></p>
</li>
<li>
<p><a href="https://github.com/hputnam/Meth_Compare/blob/master/code/MethCompare_methylKit_analysis.R">https://github.com/hputnam/Meth_Compare/blob/master/code/MethCompare_methylKit_analysis.R</a> coral methylation comparison of methods. <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/hputnam/Meth_Compare" /></p>
</li>
<li>
<p><a href="https://github.com/epigeneticstoocean/paper-gonad-meth/blob/master/code/04-methylkit.Rmd">https://github.com/epigeneticstoocean/paper-gonad-meth/blob/master/code/04-methylkit.Rmd</a> eastern oyster OA work <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/epigeneticstoocean/paper-gonad-meth" /></p>
</li>
</ul>
<h3 id="diagram_1">Diagram</h3>
<p><img alt="image" src="https://user-images.githubusercontent.com/17264765/131020085-f32e8a51-9a29-474c-aa56-2fa599e006d9.png" /> <em>Flowchart of possible operations by methylKit. A summary of the most importantmethylKit features is shown in a flow chart. It depicts the main features of methylKitand the sequential relationship between them. The functions that could be used for thosefeatures are also printed in the boxes.</em> - Figure and caption from <a href="https://doi.org/10.1186/gb-2012-13-10-r87">Akalin et al. 2012</a></p>
<hr />
<h2 id="characterizing-gene-level-methylation">Characterizing Gene Level Methylation</h2>
<h3 id="use-cases-from-our-lab_3">Use cases from our lab</h3>
<p><a href="https://sr320.github.io/gene-meth/">https://sr320.github.io/gene-meth/</a></p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit"], "search": "../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.94c44541.min.js"></script>
      
    
  </body>
</html>